### PR TITLE
Add types to actions/general.js

### PR DESCRIPTION
**Pull request by pepf**
#### Summary
Add flow types to actions/general.js

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/8996

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

#### Test Information
This PR was tested on: [macOS High sierra, 10.13.5]

### CODE REVIEW

File: src/actions/general.js [hunk 1]
```diff
@@ -115,7 +123,8 @@ export function getLicenseConfig() {
     );
 }
 
-export function logClientError(message, level = 'ERROR') {
+type logLevel = 'ERROR' | 'WARNING' | 'INFO' // I guess?
```
Comment (jespino) [inline src/actions/general.js]: this type, probably, will be used in `src/client/client4.js` file too, so I would move it somewhere in `types` directory

File: src/actions/general.js [hunk 2]
```diff
@@ -9,11 +10,18 @@ import {loadRolesIfNeeded} from './roles';
 import {logError} from './errors';
 import {batchActions} from 'redux-batched-actions';
 
-export function getPing() {
-    return async (dispatch, getState) => {
-        dispatch({type: GeneralTypes.PING_REQUEST}, getState);
+import type {GeneralState} from '../types/general';
+import type {GetStateFunc, DispatchFunc, ActionFunc} from '../types/actions';
+type response = {
```
Comment (jespino) [inline src/actions/general.js]: Probably I would move it to `types`, and change the name, If this is a generic response from the cline, I would name it as "GenericClientResponse" or something like that. We can change it in the future and use specific responses for each client4.js responses.

File: src/actions/general.js [hunk 3]
```diff
@@ -39,17 +47,17 @@ export function getPing() {
...
 
         return {data: true};
     };
 }
 
-export function getClientConfig() {
-    return async (dispatch, getState) => {
-        dispatch({type: GeneralTypes.CLIENT_CONFIG_REQUEST}, getState);
+export function getClientConfig(): ActionFunc {
+    return async (dispatch: DispatchFunc, getState) => {
```
Comment (jespino) [inline src/actions/general.js]: I miss the type of getState

File: src/actions/general.js [hunk 4]
```diff
@@ -78,9 +86,9 @@ export function getClientConfig() {
     };
 }
 
-export function getDataRetentionPolicy() {
-    return async (dispatch, getState) => {
-        dispatch({type: GeneralTypes.DATA_RETENTION_POLICY_REQUEST}, getState);
+export function getDataRetentionPolicy(): ActionFunc {
+    return async (dispatch: DispatchFunc, getState) => {
```
Comment (jespino) [inline src/actions/general.js]: I miss the type of getState

File: src/utils/key_mirror.js [hunk 5]
```diff
@@ -28,19 +28,20 @@
...
  */
-export default function keyMirror(obj: {[string]: null}): {[string]: string} {
+
+export default function keyMirror(obj: {[string]: null}) {
     if (!(obj instanceof Object && !Array.isArray(obj))) {
         throw new Error('keyMirror(...): Argument must be an object.');
     }
 
-    const ret = {};
+    type Out = {[string]: $Keys<typeof obj>;}
```
Comment (jespino) [inline src/utils/key_mirror.js]: Interesting :)

File: src/actions/general.js (hunk 1 above)
Comment (pepf) [reply src/actions/general.js]: Time to make a client4 types file!
(and immediately change some work I did in the other PR 🙈)

Comment (jespino) [reply src/actions/general.js]: Client4.js will be a big amount of work, If you want, go for it, and if you need any kind of help, let me know and I'll try to help as much as possible. :)

**Comment by jespino** [comment]
Thanks @pepf !! :sparkles:

**Pull request merged**